src/mono/wasi/mono-wasi-driver/driver.c

```diff
@@ -0,0 +1,624 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```

Comment:
This file is mostly the same as `driver.c` from `src/mono/wasm`, but with a bunch of additional helpers for locating and loading assemblies to simplify the samples. Later we might want to factor out and share some common parts.